packages/edit-navigation/src/components/menu-locations-editor/index.js

```diff
@@ -1,3 +1,167 @@
+/**
+ * External dependencies
+ */
+import { map, flatMap, remove, find, pick, groupBy } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import {
+	Button,
+	Panel,
+	PanelBody,
+	SelectControl,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MenuSelectControl = ( { onChange, menusList, location } ) => {
+	const [ menuId, setMenuId ] = useState( false );
+
+	useEffect( () => {
+		setMenuId( location.menu );
+	}, [ location.menu ] );
+
+	return (
+		<SelectControl
+			options={ menusList }
+			value={ menuId }
+			onChange={ ( selectedMenuId ) => {
+				onChange( menuId, {
+					location: location.name,
+					menu: parseInt( selectedMenuId ),
+				} );
+				setMenuId( selectedMenuId );
+			} }
+		/>
+	);
+};
+
 export default function MenuLocationsEditor() {
-	return <>Menu locations editor</>;
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus() );
+
+	const [ menuLocations, setMenuLocations ] = useState( false );
+	const [ locationsData, setLocationsData ] = useState( {} );
+	const { saveMenu } = useDispatch( 'core' );
+
+	const kind = 'root';
+	const name = 'menuLocation';
+	const entities = useSelect( ( select ) =>
+		select( 'core' ).getEntitiesByKind( kind )
+	);
+	const entity = find( entities, { kind, name } );
+
+	const getMenuLocations = async () => {
+		const path = `${ entity.baseURL }`;
+		let apiLocations = await apiFetch( {
+			path,
+			method: 'GET',
+		} );
+		apiLocations = flatMap( apiLocations, ( value ) => value );
+		return apiLocations;
+	};
+
+	const setLatestLocations = async () => {
+		const latestLocations = await getMenuLocations();
+		setMenuLocations( latestLocations );
+	};
+
+	useEffect( () => {
```

Comment:
Couldn't this just be. 
```
useEffect( setLatestLocations, [] );
```

Reply:
apparently no because setLatestLocations is an async function and useEffect doesn't like to return a promise.